### PR TITLE
Add travis CI and preliminary tests

### PR DESCRIPTION
**Pull request by ayan-b**
Related to #11

### CODE REVIEW

**Comment by ayan-b** [comment]
@yunhailuo some possible improvements:
- We can put all the scripts inside a folder called `xena-GDC-ETL`, so that it is easier to write tests for them and check coverage. This will also ease the process of releasing to pypi a lot. 
- So, in my opinion it would be better if the directory structure is something like:
 ```
xena-GDC-ETL
|- tests
|- docs
|- xena-GDC-ETL
|- .travis.yml
|- setup.py
|- ...
```

**Comment by yunhailuo** [comment]
Overall, suggestions all look good. I would prefer doing this step by step (commit) so it's clear where we should debug if anything goes wrong. I won't do coverage for now since this project is not big for now. A overall structure change and adding tests are must and they are clearly big flaws I made. So as mentioned above, I'd say separate structure change and travis. And since travis may need quite some changes after structure changes, I would do structure change in another PR first. Then come back to fix tests and/or adding more tests in this PR or new PRs.

**Comment by yunhailuo** [comment]
OK. I create a PR #13 for structural changes. It should work as far as I know for now. But I want to give myself more time for reviewing and testing before merging. Feel free to check it out and play with it though.

**Comment by ayan-b** [comment]
Ok, I will separate out the commits.

**Comment by ayan-b** [comment]
I have removed `codecov` and separated out into two commits. Will rebase this PR once #13 is merged.

**Comment by ayan-b** [comment]
I have removed `codecov` and separated out into two commits. Will rebase this PR once #13 is merged.

**Comment by ayan-b** [comment]
@yunhailuo , I have added some more tests. Please review. `gdc` module api is up to date with upstream.

**Comment by ayan-b** [comment]
Will squash the commits once this PR is approved.

**Comment by yunhailuo** [comment]
@ayan-b I just merged #13 Could you please resolve the conflict and make sure Travis test can pass?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - lint
+
+stage: test
+
+script: pytest
+
+before_install:
+  - if [[ $TRAVIS_PYTHON_VERSION == "2.6" ]]; then pip install flake8==2.6.2; fi
+  - pip install -r requirements.txt
+  - pip install pytest flake8
```
Comment (yunhailuo) [inline .travis.yml]: Will pip install this repo and add flake8 to `tests_require` more reasonable here?

Comment (ayan-b) [reply .travis.yml]: I think so. If we need more additional dependencies, this line will get quite longer. Should I make a commit?

Comment (yunhailuo) [reply .travis.yml]: Yep. I would consider this as a part of your travis test feature.

**Comment by ayan-b** [comment]
PR should be good to go now. :tada:

**Comment by ayan-b** [comment]
@yunhailuo, once this PR is merged, I will send PR on code cleanup.

**Comment by ayan-b** [comment]
@yunhailuo, please review and let me know if some modifications are required. :smile:

**Pull request merged**